compute/sqlserver/powershell/create-sql-instance-availability-group-prerequisites.ps1

```diff
@@ -0,0 +1,384 @@
+        -InstallDNS:$true `
+        -NoRebootOnCompletion:$false `
+        -SafeModeAdministratorPassword $admin_pwd `
+        -Force:$true
+
+       Write-Host "$(Get-Date) Rebooting the instance"
+     }
+     else {
+       Write-Host "  The server instance is already a member of a domain"
+     }
+  } -ArgumentList $admin_pwd, $DomainName, $DomainMode, $DatabasePath, $LogPath, `
+    $SysvolPath, $ForestMode
+
+
+  # Remove the session object
+  Remove-PSSession $session
+  Remove-Variable session
+
+
+  ################################################################################
+  # Wait until the instances is restarted
+  ################################################################################
+  $creation_status = Get-GceInstance -zone $zone -Name $domain_cntr -SerialPortOutput | 
+    Select-String -Pattern 'Finished running shutdown scripts' -Quiet
+  while (!($creation_status)) {
+    Write-Host "$(Get-Date) Waiting for instance $domain_cntr to restart"
+    Start-Sleep 15
+
+    $creation_status = Get-GceInstance -zone $zone -Name $domain_cntr -SerialPortOutput | 
+      Select-String -Pattern 'Finished running shutdown scripts' -Quiet
+  }
+
+  # Wait a minute to make sure all AD services are running before we continue
+  Start-Sleep 60
+
+  ################################################################################
+  # Create a remote session to the domain controller
+  ################################################################################
+  $session_options = New-PSSessionOption -SkipCACheck -SkipCNCheck `
+    -SkipRevocationCheck
+  $session = New-PSSession -ComputerName $ip_address -UseSSL `
+    -Credential $credential -SessionOption $session_options
+
+
+  ################################################################################
+  # Change the password for the domain administrator
+  ################################################################################
+  Write-Host "$(Get-Date) Change password for Administrator and sql.service"
+  Invoke-Command -Session $session -ScriptBlock {
+    param($admin_pwd, $sql_pwd)
+  
+    Set-ADAccountPassword `
+      -Identity "Administrator" `
+      -Reset `
+      -NewPassword $admin_pwd
+
+  } -ArgumentList $admin_pwd, $sql_pwd
+
+
+  # Remove the session object
+  Remove-PSSession $session
+  Remove-Variable session
+
+} else {
+  Write-Host "  The instance: $domain_cntr already exist"
+}
+
+
+Write-Host " "
+Write-Host "IMPORTANT: You are now ready to run the script  create-sql-instance-availability-group.ps1"
+Write-Host "           It is recommended to run it from an instance that is a member of the domain: $DomainName"
+Write-Host "           if you want to automatically setup the SQL Server Availability Group."
+
+# Remove firewall rule to allow current IP
+#Get-GceFirewall -FirewallName "allow-winrm-current-ip" | Remove-GceFirewall
+
+##### Cleanup - Run with caution as everything will be deleted #####

```

Comment:
Agree. I'll remove it from here and put it in a new script with the same name but with "-cleanup.ps1" at the end.